lib/reporter.express.js

```diff
@@ -159,15 +162,20 @@ var configureExpressApp = function (app, reporter, definition) {
     next()
   })
 
-  reporter.emit('before-express-configure', app)
+  reporter.emit('before-express-configure', baseApp)
 
-  routes(app, reporter)
+  routes(baseApp, reporter)
 
-  reporter.emit('express-configure', app)
+  reporter.emit('express-configure', baseApp)
 
   const handleErrorMiddleware = handleError(reporter)
 
-  app.use((err, req, res, next) => handleErrorMiddleware(req, res, err))
+  baseApp.use((err, req, res, next) => handleErrorMiddleware(req, res, err))
+
+  if (reporter.options.subPath) {
+    app.use(reporter.options.subPath, baseApp)
+    reporter.express.app = app

```

Comment:
finally, as i say in original review, we don't like that `reporter.express.app` is changed to different things. so this line should be removed, we think that with `baseApp` is enough and it should be the only thing exposed (both in events and in reporter.express.app)

Reply:
If we remove this line, app will never be run ( it would be baseapp ) , and the subpath will not be effective ( cf [tests](https://github.com/jsreport/jsreport-express/pull/9/files#diff-6930fff1c7088d72e387e45fde3e6afaR173) ) 
Furthermore, if we use subPath in conf, we want to plugin routes to also be provided by the subPath, so we thinx that app shoulb be exposed. 

We dont see how to change the code as you ask, without breaking the feature and the new tests

Reply:
i checked locally and you are right, removing that line makes the PR not work. that is because there are other changes needed in code, i have done those changes locally but now we have concerns about this new option `subPath` ending too similar to `appPath`. for example we have in extensions code like [this](https://github.com/jsreport/jsreport-studio/blob/master/lib/studio.js#L23)

Reply:
i checked locally and you are right, removing that line makes the PR not work. that is because there are other changes needed in code, i have done those changes locally but now we have concerns about this new option `subPath` ending too similar to `appPath`. for example we have in extensions code like [this](https://github.com/jsreport/jsreport-studio/blob/master/lib/studio.js#L23), those extensions will need to add another condition that does the same but now using `subPath` too, so we are evaluating the case more in depth and checking if we can find another solution to your original problem that does not require duplicating logic. stay tuned